agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -200,48 +197,43 @@ public String getTsaLicensePath(InstanceId instanceId, TerracottaServer terracot
   }
 
   public boolean installTms(InstanceId instanceId, String tmsHostname, Distribution distribution, License license,
-                            TmsServerSecurityConfig tmsServerSecurityConfig, String kitInstallationName,
-                            TerracottaCommandLineEnvironment tcEnv, Collection<String> hostNames) {
+                            String kitInstallationName,
+                            TerracottaCommandLineEnvironment tcEnv, String hostName, String kitInstallationPath) {
     TmsInstall tmsInstall = tmsInstalls.get(instanceId);
     if (tmsInstall != null) {
       logger.debug("Kit for " + tmsHostname + " already installed");
       tmsInstall.addTerracottaManagementServer();
       return true;
-    } else {
-      logger.debug("Attempting to install kit from cached install for " + tmsHostname);
-      RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
 
-      if (kitManager.isKitAvailable()) {
-        File kitDir = kitManager.installKit(license, hostNames);
-        File workingDir = kitManager.getWorkingDir().toFile();
+    } else {
 
-        File tmcProperties = new File(kitDir, "/tools/management/conf/tmc.properties");
-        if (tmsServerSecurityConfig != null) {
-          enableTmsSecurity(tmcProperties, tmsServerSecurityConfig);
-        }
-        tmsInstalls.put(instanceId, new TmsInstall(distribution, kitDir, workingDir, tcEnv));
-        return true;
-      } else {
+      Optional<Dirs> dirs = Dirs.discover(instanceId, hostName, distribution, license, kitInstallationName, kitInstallationPath);
```

Comment:
are you sure removing this code  is not impacting the tests when not using the kitsInstallationDir property?

Reply:
oh right! I will change! The logic needs to stay the same for tests not using that prop.